### ISSUE TITLE
Uncaught exception when using left arrow key while filtering choices

### ISSUE THREAD

**Issue by cswilliams** [piotrmurach/tty-prompt#155]
### Describe the problem
I'm seeing an uncaught exception when using the left arrow key while filtering a set of choices.
### Steps to reproduce the problem
```
require 'tty-prompt'
TTY::Prompt.new.select("Choose:", cycle: true, filter: true, per_page: 10) do |menu|
  menu.choice "a"
  menu.choice "ab"
end
```
1) Run the code above
2) Type the filter phrase `ab`, and then press the left arrow key. 
3) The program will throw an exception and exit.

### Actual behaviour
The following error is raised:
```
Traceback (most recent call last):
	23: from test.rb:2:in `<main>'
	22: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt.rb:286:in `select'
	21: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt.rb:249:in `invoke_select'
	20: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:240:in `call'
	19: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/forwardable.rb:230:in `subscribe'
	18: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:118:in `subscribe'
	17: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:241:in `block in call'
	16: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:423:in `render'
	15: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/forwardable.rb:230:in `read_keypress'
	14: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:190:in `read_keypress'
	13: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:476:in `trigger_key_event'
	12: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:422:in `trigger'
	11: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/publisher.rb:42:in `broadcast'
	10: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/set.rb:338:in `each'
	 9: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/set.rb:338:in `each_key'
	 8: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/publisher.rb:43:in `block in broadcast'
	 7: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/registration/object.rb:18:in `broadcast'
	 6: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/broadcasters/send_broadcaster.rb:5:in `broadcast'
	 5: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/broadcasters/send_broadcaster.rb:5:in `public_send'
	 4: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:337:in `keyleft'
	 3: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:268:in `search_choice_in'
	 2: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:268:in `find'
	 1: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:268:in `each'
/Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:268:in `block in search_choice_in': undefined method `disabled?' for nil:NilClass (NoMethodError)
```
### Expected behaviour
Expected not to throw an exception.

### Describe your environment

* OS version: Mac OSX 10.15.7
* Ruby version: 2.6.6
* TTY version: 0.22.0

**Comment by cswilliams** [comment]
Slightly different bug (although I'm guessing it's related, but, if not, happy to file a separate issue): 

```
require 'tty-prompt'
TTY::Prompt.new.select("Choose:", cycle: true, filter: true, per_page: 10) do |menu|
  menu.choice 'US 2020-12-10 21:02:44'
  menu.choice 'UK 2020-12-10 20:02:36'
  menu.choice 'US 2020-12-10 17:03:14'
  menu.choice 'UK 2020-12-10 16:02:13'
  menu.choice 'US 2020-12-10 14:02:54'
  menu.choice 'UK 2020-12-10 13:02:31'
  menu.choice 'US 2020-12-10 09:03:06'
  menu.choice 'UK 2020-12-10 08:02:40'
  menu.choice 'US 2020-12-10 06:02:34'
  menu.choice 'UK 2020-12-10 05:02:19'
  menu.choice 'US 2020-12-10 01:03:18'
  menu.choice 'UK 2020-12-10 00:03:45'
  menu.choice 'US 2020-12-09 21:03:02'
  menu.choice 'UK 2020-12-09 20:02:34'
  menu.choice 'US 2020-12-09 17:03:04'
  menu.choice 'UK 2020-12-09 16:03:09'
  menu.choice 'US 2020-12-09 14:02:28'
end
```
Then run the program, type in the phrase `US` and then press the right arrow key **_twice_**.

This following backtrace is then thrown:
```
Traceback (most recent call last):
	19: from test.rb:2:in `<main>'
	18: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt.rb:286:in `select'
	17: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt.rb:249:in `invoke_select'
	16: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:240:in `call'
	15: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/forwardable.rb:230:in `subscribe'
	14: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:118:in `subscribe'
	13: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:241:in `block in call'
	12: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:423:in `render'
	11: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/forwardable.rb:230:in `read_keypress'
	10: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:190:in `read_keypress'
	 9: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:476:in `trigger_key_event'
	 8: from /Users/user/.rvm/gems/ruby-2.6.6/gems/tty-reader-0.9.0/lib/tty/reader.rb:422:in `trigger'
	 7: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/publisher.rb:42:in `broadcast'
	 6: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/set.rb:338:in `each'
	 5: from /Users/user/.rvm/rubies/ruby-2.6.6/lib/ruby/2.6.0/set.rb:338:in `each_key'
	 4: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/publisher.rb:43:in `block in broadcast'
	 3: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/registration/object.rb:18:in `broadcast'
	 2: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/broadcasters/send_broadcaster.rb:5:in `broadcast'
	 1: from /Users/user/.rvm/gems/ruby-2.6.6/gems/wisper-2.0.1/lib/wisper/broadcasters/send_broadcaster.rb:5:in `public_send'
/Users/user/.rvm/gems/ruby-2.6.6/gems/tty-prompt-0.22.0/lib/tty/prompt/list.rb:311:in `keyright': undefined method `+' for nil:NilClass (NoMethodError)
```

**Comment by piotrmurach** [comment]
Released `v0.23.0` with fixes.

**Comment by cswilliams** [comment]
Hi Piotr,

I just tested your fix and everything works perfectly now! Thanks so much for the quick fixes and for making this fantastic library!
